src/base/static/components/place-detail/place-detail-editor.js

```diff
@@ -0,0 +1,218 @@
+        .reduce((newValidationErrors, invalidField) => {
+          return newValidationErrors.add(
+            invalidField.get(constants.FIELD_STATE_VALIDITY_MESSAGE_KEY)
+          );
+        }, new Set());
+
+      if (newValidationErrors.size === 0) {
+        this.props.onUpdate(this.state.fields);
+      } else {
+        this.setState({
+          formValidationErrors: newValidationErrors,
+          showValidityStatus: true,
+        });
+        scrollTo(this.props.container, 0, 300);
+      }
+    });
+    emitter.addListener("place-model:remove", () => {
+      this.props.onRemove();
+    });
+
+    // Set the initial geometry style.
+    if (this.props.placeModel.get(constants.GEOMETRY_STYLE_PROPERTY_NAME)) {
+      this.props.onGeometryStyleChange(
+        this.props.placeModel.get(constants.GEOMETRY_STYLE_PROPERTY_NAME)
+      );
+    }
+  }
+
+  getMapDrawingToolbarState(fieldConfig) {
+    return fieldConfig.type === constants.MAP_DRAWING_TOOLBAR_TYPENAME
+      ? {
+          layerView: this.props.layerView,
+          initialPanel:
+            constants.GEOMETRY_EDITOR_TOOL_MAPPINGS[
+              this.props.placeModel
+                .get(constants.GEOMETRY_PROPERTY_NAME)
+                .get(constants.GEOMETRY_TYPE_PROPERTY_NAME)
+            ],
+          initialGeometryType:
+            constants.GEOMETRY_TYPE_MAPPINGS[
+              this.props.placeModel
+                .get(constants.GEOMETRY_PROPERTY_NAME)
+                .get(constants.GEOMETRY_TYPE_PROPERTY_NAME)
+            ],
+          existingLayer: this.props.layerView.layer,
+          existingColor: this.props.placeModel
+            .get(constants.GEOMETRY_STYLE_PROPERTY_NAME)
+            .get("color"),
+          existingOpacity: this.props.placeModel
+            .get(constants.GEOMETRY_STYLE_PROPERTY_NAME)
+            .get("opacity"),
+          existingFillColor: this.props.placeModel
+            .get(constants.GEOMETRY_STYLE_PROPERTY_NAME)
+            .get("fillColor"),
+          existingFillOpacity: this.props.placeModel
+            .get(constants.GEOMETRY_STYLE_PROPERTY_NAME)
+            .get("fillOpacity"),
+          selectedMarkerIndex:
+            fieldConfig.content &&
+            fieldConfig.content.findIndex(
+              marker =>
+                marker.url ===
+                this.props.placeModel
+                  .get(constants.GEOMETRY_STYLE_PROPERTY_NAME)
+                  .get(constants.ICON_URL_PROPERTY_NAME)
+            ),
+        }
+      : {};
+  }
+
+  componentDidMount() {
+    // TODO: Replace this.
+    new Spinner(Shareabouts.smallSpinnerOptions).spin(
+      document.getElementsByClassName("place-detail-editor__spinner")[0]
+    );
+  }
+
+  componentWillUnmount() {
+    emitter.removeAllListeners("place-model:update");
+    emitter.removeAllListeners("place-model:remove");
+  }
+
+  onFieldChange(fieldName, fieldStatus, isInitializing) {
```

Comment:
I'd recommend using named args here, as described in previous comments.

Also, why are we calling the arg `fieldStatus` and not `fieldValue`?

Reply:
Technically `fieldStatus` here is made up of several components: the actual field value, its validity status, a render key (where necessary), and a validity message. I was thinking the term "status" was a better description of what this variable is.